File: cryptography/rot13_cipher/rot13.py
""" USAGE

plaintext = 'This is unencrypted sample text'
ciphertext = rot13(plaintext)

or

ciphertext = 'Guvf vf harapelcgrq fnzcyr grkg'
plaintext = rot13(ciphertext)
"""

def rot13(string):
    result = ''
    for ch in string:
            ascii_value = ord(ch)
            if ch == ' ':
                    result += ' '
            elif ascii_value >= 65 and ascii_value <= 90:
                    if ascii_value + 13 > 90:
                            result += chr(ascii_value - 13)
                    else:
                            result += chr(ascii_value + 13)	
            elif ascii_value >= 97 and ascii_value <= 122:	
                    if ascii_value + 13 > 122:
                                    result += chr(ascii_value - 13)	
                    else:
                            result += chr(ascii_value + 13)		
    return result

File: cryptography/rot13_cipher/js/rot13.js
/* USAGE
plaintext = 'This is unencrypted sample text';
ciphertext = rot13(plaintext);

or

ciphertext = 'Guvf vf harapelcgrq fnzcyr grkg';
plaintext = rot13(ciphertext);
*/

function rot13(text){
  return text.replace(/[a-z]/gi, function(char){
    return String.fromCharCode((char<="Z"?90:122)>=(char=char.charCodeAt(0)+13)?char:char-26);
  });
} 
